### PR TITLE
[Doc]Add tips for troubleshooting a pipeline

### PR DESCRIPTION
**Pull request by karenzone**
Pipelines vary wildly, and it's nearly impossible to offer prescriptive steps for troubleshooting. Perhaps we can help by offering some general guidelines, things to try, etc. 

This is a basic start. I'm hoping that we can expand with more tips and guidelines now and over time.

PREVIEW: http://logstash_11545.docs-preview.app.elstc.co/guide/en/logstash/master/troubleshooting.html#ts-pipeline

### CODE REVIEW

**Comment by karenzone** [comment]
@bczifra  Thanks for the suggestion. @andsel Thanks for the tips so far.
I welcome ideas from you and from others to expand this section.

File: docs/static/troubleshooting.asciidoc [hunk 1]
```diff
@@ -139,9 +137,38 @@ For general performance tuning tips and guidelines, see <<performance-tuning>>.
...
+For basic pipelines, this configuration could be enough to make the problem show itself.
 
+[source,ruby]
+-----
+input {stdin{}} output {stdout{}}
+-----
+
+To better identify the offending pipeline, {ls} could separate its logs by pipeline.
+To switch on the log per pipeline feature is sufficient to enable `pipeline.separate_logs` in your `logstash.yml`
+or passing the `-Dls.pipeline.separate_logs=true` on command line.
```
Comment (yaauie) [inline docs/static/troubleshooting.asciidoc]: I'm not sure this is quite right, or if we even _can_ change this setting from the command line.

Although we use the Java System Property `ls.pipeline.separate_logs` in the log4j templates (and one would ordinarily set Java System Properties with `-Dproperty=value`), it looks like on startup we stomp on the existing value of that property without first reading it, stetting to to be the value held in `pipeline.separate_logs` from `logstash.yml`:

> ~~~ ruby
>     java.lang.System.setProperty("ls.pipeline.separate_logs", setting("pipeline.separate_logs").to_s)
> ~~~
> -- [logstash-core/lib/logstash/runner.rb:257@2925b491](https://github.com/elastic/logstash/blob/2925b491f9a68afc0277e35bdd6fe07ea17b4eeb/logstash-core/lib/logstash/runner.rb#L257)

Comment (andsel) [reply docs/static/troubleshooting.asciidoc]: @yaauie you are right, many thanks to point it out. Removed and rephrased a little bit

**Comment by karenzone** [comment]
Any further comments or additions?  We can expand this question later if needed.

**Comment by karenzone** [comment]
@andsel Any further comments or additions? We can expand this question later if needed.

**Comment by andsel** [comment]
LGTM

**Pull request closed without merging**